doc/specs/stdlib_selection.md

```diff
@@ -0,0 +1,364 @@
+In this case the input array is not modified, but the user must provide an
+input index array with the same size as `array`, having unique indices from
+`1:size(array)`, which is modified instead. On return the index array is modified
+such that `all(array(index(1:k)) <= array(index(k)))` and `all(array(k) <= array(k+1:size(array)))`.
+The user can optionally specify `left` and `right` indices to constrain the search
+for the kth-smallest value. This can be useful if you have previously called `arg_select`
+to find a smaller or larger rank (that will have led to partial sorting of
+`index`, thus implying some constraint on the location).
+
+#### Licensing
+
+The Fortran Standard Library is distributed under the MIT
+License. However components of the library may be based on code with
+additional licensing restrictions. In particular `select` and `arg_select`
+were derived by modifying a matlab implementation of "qselect" by Manolis
+Lourakis, https://www.mathworks.com/matlabcentral/fileexchange/68947-qselect
+Below is the licence of the matlab qselect
+
+Copyright (c) 2018, Manolis Lourakis
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+* Redistributions of source code must retain the above copyright notice, this
+  list of conditions and the following disclaimer.
+
+* Redistributions in binary form must reproduce the above copyright notice,
+  this list of conditions and the following disclaimer in the documentation
+  and/or other materials provided with the distribution
+* Neither the name of Foundation for Research and Technology - Hellas nor the
+  names of its contributors may be used to endorse or promote products
+  derived from this software without specific prior written permission.
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+### Specifications of the `stdlib_selection` procedures
+
+#### `select` - find the kth smallest value in an input array
+
+##### Status
+
+Experimental
+
+##### Description
+
+Returns the k-th smallest value of array `a(:)`, and also partially sorts `a(:)`
+such that `all(a(1:k) <= a(k))`  and `all(a(k) <= a((k+1):size(a)))`
```

Comment:
I suggest to rename the array `a` to `array`.

Reply:
I've implemented this in a new commit.